### PR TITLE
For #11905 - Filter out Pocket stories with missing image path

### PR DESCRIPTION
**Pull request by Mugurell**
### Pull Request checklist

- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/main/docs/changelog.md) or does not need one
- [x] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/main/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Milestone**: Make sure issues closed by this pull request are added to the [milestone](https://github.com/mozilla-mobile/android-components/milestones) of the version currently in development.
- **Breaking Changes**: If this is a breaking change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### CODE REVIEW

File: components/service/pocket/src/main/java/mozilla/components/service/pocket/api/PocketJSONParser.kt [hunk 1]
```diff
@@ -54,6 +56,30 @@ internal class PocketJSONParser {
...
+     */
+    @Throws(JSONException::class)
+    @VisibleForTesting
+    internal fun getValidImageUrl(jsonStory: JSONObject): String {
+        val imageUrl = jsonStory.getString(JSON_STORY_IMAGE_URL_KEY)
+        val imagePath = imageUrl.substringAfterLast("/")
+
+        if (imagePath == "null") {
+            throw JSONException("Null image path")
+        }
```
Comment (Mugurell) [inline components/service/pocket/src/main/java/mozilla/components/service/pocket/api/PocketJSONParser.kt:78]: The solution from this patch is hardcoded for a specific scenario - if the imageUrl ends with "/null".
(we were already dropping stories with the entire imageUrl being `null`).
Will try to get someone from Pocket to review on whether this is enough or there might be other scenarios we'd need to account for.

**Review by csadilek — APPROVED** [review]
I think this is fine as a workaround / fallback, but the underlying issue should be fixed here so we don't receive `/null`, otherwise we'd essentially just hide this problem.

**Comment by Mugurell** [comment]
> I think this is fine as a workaround / fallback, but the underlying issue should be fixed here so we don't receive `/null`, otherwise we'd essentially just hide this problem.

This exact will first be tried to be fixed on Pocket's side.

We could have a placeholder drawable / link in Fenix for other such situations, checked with Kevin about that.

**Comment by Mugurell** [comment]
Closing this in favor of https://github.com/mozilla-mobile/android-components/pull/11918 to support Pocket's updates.
Whether Fenix should use it's own fallback image can be assessed separately.

**Pull request closed without merging**